### PR TITLE
Fairmotion Character Positioning via MOTION MouseMode

### PR DESCRIPTION
**Pull request by JuanTheEngineer**
### Goal
With this PR merged, the Fairmotion viewer will have an additional mouse interaction mode called "MOTION".

### Functionality
This mode allows users to select a loaded Fairmotion character and click to a desired location for the character and its motion. It also allows users to set rotation around the floor-normal axis by scrolling the mouse wheel with fine-grain rotating using the Shift key. These are useful features for positioning the character in the scene. 

It is works in parallel with the Key Frame and Trajectory Object Previews, which gives the user the ability to foresee scene collision throughout the timeline of the motion for improved character placement choice.

Currently the Fairmotion viewer supports one character at a time and this PR feature is lightly coupled to that fact. In a future PR multiple character loading will be supported and this MOTION mode with expand to the same degree.

### UI Mouse Commands
Here is a snippet of the Command Help Text to describe the utilities of this PR
```
In MOTION mode (with 'enable-physics'):
    LEFT:
        Click a Fairmotion character to set it as selected or clcik anywhere else to deselect.
    RIGHT (With selected Fairmotion character):
        Click anywhere on the scene to translate a selected Fairmotion character to the clicked location.
    WHEEL (with selected Fairmotion character):
        Rotate the orientation of a selected Fairmotion character along an axis normal to the floor of the scene.
        (+ SHIFT): for fine-grained control
```

### Preview
Here is a preview of the feature in action where the feature is used to place the character's dribbling motion in the tight hallway:

https://user-images.githubusercontent.com/59671006/139751508-b0673e74-14cc-4407-a34f-14545885a2f6.mp4

## How Has This Been Tested

The majority of this feature lives in the Fairmotion viewer application and was tested by running that application.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/fairmotion_interface.py [hunk 1]
```diff
@@ -45,7 +44,8 @@ def __init__(
         LoggingContext.reinitialize_from_env()
         self.sim = sim
         self.art_obj_mgr = self.sim.get_articulated_object_manager()
-        self.model: habitat_sim.physics.ManagedArticulatedObject = None
+        self.rgd_obj_mgr = self.sim.get_rigid_object_manager()
+        self.model: phy.ManagedArticulatedObject = None
```
Comment (Skylion007) [inline examples/fairmotion_interface.py:48]:
```suggestion
        self.model: Optional[phy.ManagedArticulatedObject] = None
```

File: examples/fairmotion_interface.py [hunk 2]
```diff
@@ -45,7 +44,8 @@ def __init__(
...
         self.motion: motion.Motion = None
```
Comment (Skylion007) [inline examples/fairmotion_interface.py:49]:
```suggestion
        self.model: Optional[phy.ManagedArticulatedObject] = None
```

File: examples/fairmotion_interface.py [hunk 3]
```diff
@@ -470,6 +463,32 @@ def cycle_model_previews(self) -> None:
...
+            return True
+
+        # checking our model
+        if self.model and obj_id is self.model.object_id:
+            return True
+
+        # checking all key frame models
+        if any(
+            obj_id in ko_ids
+            for ko_ids in [i.link_object_ids.keys() for i in self.key_frame_models]
```
Comment (Skylion007) [inline examples/fairmotion_interface.py:482]: Nested comprehension should probably be a set comprehension
```suggestion
            for ko_ids in {i.link_object_ids.keys() for i in self.key_frame_models}
```

**Review by Skylion007 — CHANGES_REQUESTED** [review]

**Comment by Skylion007** [comment]
FYI, just updated the Magnum color bindings to make things a bit easier for you: https://github.com/mosra/magnum-bindings/commit/c50a2f68514495b6ca125297f1c7a39d02d5878c . Will need to bump the submodule.

File: examples/fairmotion_interface.py [hunk 4]
```diff
@@ -435,23 +452,19 @@ def define_preview_points(joint_names: List[str]) -> List[mn.Vector3]:
 
         # TODO: This function is not working. It is supposed to produce a gradient
         #       from RED to YELLOW to GREEN but it is producing a black solely
-        colors = [
-            mn.Color3(255, 0, 0),
-            mn.Color3(255, 255, 0),
-            mn.Color3(0, 255, 0),
-        ]
+        colors = [mn.Color3(255, 0, 0), mn.Color3(255, 255, 0), mn.Color3(0, 255, 0)]
```
Comment (Skylion007) [inline examples/fairmotion_interface.py:455]:
```suggestion
        colors = [mn.Color3(1.0, 0, 0), mn.Color3(1.0, 1.0, 0), mn.Color3(0, 1.0, 0)]
```

File: examples/fairmotion_interface.py (hunk 3 above)
Comment (Skylion007) [reply examples/fairmotion_interface.py:502]: Ah, I misinterepreted what "in" means in this context. No reason to store a temporarly list like that though, better just to chain the generators.

```suggestion
            for ko_ids in (i.link_object_ids.keys() for i in self.key_frame_models)
```

File: examples/fairmotion_interface.py (hunk 4 above)
Comment (JuanTheEngineer) [reply examples/fairmotion_interface.py:455]: This is weird, I am pretty sure I pushed this line to be using the bindings `Color3.red()` etc.

**Review by aclegg3 — APPROVED** [review]
Great video in description. Let's improve the UI description also. Be clear about which click and keys to use. Maybe use a bullet list similar to the help text.

When you re-defined the help text for this application you dropped the mouse legend. Let's add that back here and document the `MOTION` mode there.

**Pull request merged**